src/generic/listctrl.cpp

```diff
@@ -3302,41 +3302,53 @@ void wxListMainWindow::OnKillFocus( wxFocusEvent &WXUNUSED(event) )
 
 void wxListMainWindow::DrawImage( int index, wxDC *dc, int x, int y )
 {
-    if ( HasFlag(wxLC_ICON) && (m_normal_image_list))
+    if ( HasFlag(wxLC_ICON) && (m_normal_images))
     {
-        m_normal_image_list->Draw( index, *dc, x, y, wxIMAGELIST_DRAW_TRANSPARENT );
+        const auto bmp = m_normal_images->GetImageBitmapFor(this, index);
+        dc->DrawBitmap(bmp, x, y);
     }
-    else if ( HasFlag(wxLC_SMALL_ICON) && (m_small_image_list))
+    else if ( HasFlag(wxLC_SMALL_ICON) && (m_small_images))
     {
-        m_small_image_list->Draw( index, *dc, x, y, wxIMAGELIST_DRAW_TRANSPARENT );
+        const auto bmp = m_small_images->GetImageBitmapFor(this, index);
+        dc->DrawBitmap(bmp, x, y);
     }
-    else if ( HasFlag(wxLC_LIST) && (m_small_image_list))
+    else if ( HasFlag(wxLC_LIST) && (m_small_images))
     {
-        m_small_image_list->Draw( index, *dc, x, y, wxIMAGELIST_DRAW_TRANSPARENT );
+        const auto bmp = m_small_images->GetImageBitmapFor(this, index);
+        dc->DrawBitmap(bmp, x, y);
     }
-    else if ( InReportView() && (m_small_image_list))
+    else if ( InReportView() && (m_small_images))
     {
-        m_small_image_list->Draw( index, *dc, x, y, wxIMAGELIST_DRAW_TRANSPARENT );
+        const auto bmp = m_small_images->GetImageBitmapFor(this, index);
+        dc->DrawBitmap(bmp, x, y);
     }
 }
 
 void wxListMainWindow::GetImageSize( int index, int &width, int &height ) const
 {
-    if ( HasFlag(wxLC_ICON) && m_normal_image_list )
+    if ( HasFlag(wxLC_ICON) && m_normal_images )
     {
-        m_normal_image_list->GetSize( index, width, height );
+        const auto size = m_normal_images->GetImageLogicalSize(this, index);
+        width = size.GetWidth();
+        height = size.GetHeight();
```

Comment:
We have an overload taking `width` and `height` directly, i.e. this could be just
```suggestion
        m_normal_images->GetImageLogicalSize(this, index, width, height);
```

(and the same thing below, of course).